src/main/java/com/diffblue/demo/ecommerce/ProductController.java

```diff
@@ -39,4 +38,16 @@ public String productList(Map<String, Object> model) {
     return "ProductList";
   }
 
+  @RequestMapping("/product/{id}")
+  public String productPage(@PathVariable("id") String id, Map<String, Object> model) {
+    Product prod = this.productRepo.findById(id);
+    if(prod != null){
+      Application.log.info("Product as string: " + prod.toString());
+      model.put("product", prod);
+      return "ProductPage";
+    }else{
+      return "redirect:/ProductList";
```

Comment:
Would it be helpful to log here the id that was being tried to the application log?

Reply:
Addressed.